flow/Net2.actor.cpp

```diff
@@ -200,6 +207,9 @@ class Net2 sealed : public INetwork, public INetworkConnections {
 	void trackMinPriority( TaskPriority minTaskID, double now );
 	void stopImmediately() {
 		stopped=true; decltype(ready) _1; ready.swap(_1); decltype(timers) _2; timers.swap(_2);
+		for ( auto& fn : stopCallbacks ) {
+			fn();
+		}
```

Comment:
It's funny that I just asked you about having a callback like this in another PR without seeing this one.

This function is setting the `stopped` flag, which isn't checked until the next time the run loop goes around, and I think that possible additional tasks could run in the meantime. Is there any risk that the user of this callback needs to clean something up that subsequently breaks whatever the run loop does next? Also, @mpilman does this scheme work for your use-case?

An alternative approach to avoid this problem if it exists is to fire the callbacks after we leave the loop in `Net2::run()`.

Reply:
Loop has been moved.  Correspondingly, I've shuffled the TLS global state destruction to use this shutdown hook.